code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingFullscreenMessageDelegate.java

```diff
@@ -104,34 +100,50 @@ public boolean overrideUrlLoad(final FullscreenMessage fullscreenMessage, final
             return false;
         }
 
+        // url decode the query parameters
+        String queryParams;
+        try {
+            queryParams = URLDecoder.decode(uri.getQuery(), StandardCharsets.UTF_8.toString());
+        } catch (final UnsupportedEncodingException exception) {
+            Log.debug(MessagingConstants.LOG_TAG, SELF_TAG,  "UnsupportedEncodingException occurred when decoding query parameters %s.", uri.getQuery());
+            return false;
+        }
+
         // Populate message data
-        final String query = uri.getQuery();
-        final Map<String, String> messageData = extractQueryParameters(query);
+        final Map<String, String> messageData = extractQueryParameters(queryParams);
 
         final MessageSettings messageSettings = fullscreenMessage.getMessageSettings();
         final Message message = (Message) messageSettings.getParent();
 
         if (!MapUtils.isNullOrEmpty(messageData)) {
             // handle optional tracking
-            final String interaction = messageData.get(MessagingConstants.QueryParameters.INTERACTION);
+            final String interaction = messageData.remove(MessagingConstants.QueryParameters.INTERACTION);
             if (!StringUtils.isNullOrEmpty(interaction)) {
                 // ensure we have the MessagingExtension class available for tracking
                 final Object messagingExtension = message.getParent();
                 if (messagingExtension != null) {
+                    Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Tracking message interaction (%s)", interaction);
                     message.track(interaction, MessagingEdgeEventType.IN_APP_INTERACT);
                 }
             }
 
             // handle optional deep link
-            final String url = messageData.get(MessagingConstants.QueryParameters.LINK);
-            if (!StringUtils.isNullOrEmpty(url)) {
-                openUrl(fullscreenMessage, url);
-            }
-
-            // handle optional javascript code to be executed
-            final String javascript = messageData.get(MessagingConstants.QueryParameters.JAVASCRIPT_QUERY_KEY);
-            if (!StringUtils.isNullOrEmpty(javascript)) {
-                message.evaluateJavascript(javascript);
+            String deeplink = messageData.remove(MessagingConstants.QueryParameters.LINK);
+            if (!StringUtils.isNullOrEmpty(deeplink)) {
+                // handle optional javascript code to be executed
+                if (deeplink.startsWith(MessagingConstants.QueryParameters.JAVASCRIPT_QUERY_KEY)) {
```

Comment:
nit: Should we also check here that interaction type is `javascript`?

Reply:
the interaction type is specified on the AJO UI so it can be any value.